ctc_lib/container_tool_list_command.go

```diff
@@ -76,32 +88,58 @@ Either implement Command.Run implementation or RunO implemetation`)
 }
 
 func (ctc *ContainerToolListCommand) printO(c *cobra.Command, args []string) error {
-	var err error
+	var commandError, totalError error
+	var totalDisplayString string
 	if ctc.StreamO != nil {
-		// Stream Objects
+		// Stream Objects only when JsonOutput = False
 		ctc.StreamO(c, args)
-		ctc.OutputList, err = ctc.ReadFromStream()
+		ctc.OutputList, commandError = ctc.ReadFromStream(!flags.JsonOutput)
 	} else {
 		// Run RunO function.
-		ctc.OutputList, err = ctc.RunO(c, args)
-		if err != nil {
-			return err
-		}
-		err = util.ExecuteTemplate(ctc.ReadTemplateFromFlagOrCmdDefault(),
-			ctc.OutputList, ctc.TemplateFuncMap, ctc.OutOrStdout())
+		ctc.OutputList, commandError = ctc.RunO(c, args)
+		LogIfErr(commandError, Log)
 	}
-	if err != nil {
-		return err
+	totalDisplayString, commandError, totalError = ctc.runTotalIfDefined()
+	LogIfErr(commandError, Log)
+	ctc.printResult(totalDisplayString)
+	LogIfErr(totalError, Log)
+
+	if commandError != nil && totalError != nil {
+		return errors.New("One or more errors")
```

Comment:
Are errors handled the same way if json output is used?  It might be worth adding some test cases for failing commands as well in another PR.

Reply:
will add tests in this PR :) Thanks @aaron-prindle.